pkg/apis/pipeline/v1alpha1/resource_types.go

```diff
@@ -130,15 +130,29 @@ type PipelineResourceList struct {
 	Items           []PipelineResource `json:"items"`
 }
 
-// ResourceFromType returns a PipelineResourceInterface from a PipelineResource's type.
-func ResourceFromType(r *PipelineResource) (PipelineResourceInterface, error) {
+// InputResourceFromType returns a PipelineResourceInterface from a PipelineResource's type.
+func InputResourceFromType(r *PipelineResource) (PipelineResourceInterface, error) {
```

Comment:
would it be crazy in Go to define a top level like map of types to `New` functions? e.g. something like:

```go
PipelineResourceTypeGit: NewGitResource,
PipelineResourceTypeImage: NewImageResource,
```

If case statements are clearer, just ignore me